### PR TITLE
Oauth helper class

### PR DESCRIPTION
**Pull request by mrben**
Hello

I've added an OAuth helper class to assist with getting a code and exchanging that for an access token, as documented [here](https://www.mixcloud.com/developers/#authorization).

```python
import mixcloud

MIXCLOUD_CLIENT_ID = 'qwerty'
MIXCLOUD_CLIENT_SECRET = 'ytrewq'
redirect_uri = 'http://localhost:8080/my-mixcloud-callback'

auth = mixcloud.MixcloudOauth(
    client_id=MIXCLOUD_CLIENT_ID, client_secret=MIXCLOUD_CLIENT_SECRET,
    redirect_uri=redirect_uri)

url = auth.authorize_url()

# Redirect to the user to the url, Mixcloud will redirect them to
# `redirect_uri` and include a `code` parameter in the query string.

access_token = auth.exchange_token(code)
m = mixcloud.Mixcloud(access_token=access_token)
print(m.user('mrben').name)
```

Hope you think this is useful.

I'm more than happy to rework it if you want it done in a different way.

Ben.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -16,6 +16,38 @@ Example
...
+3. Collect the `code` query string parameter from the redirect
+4. Exchange the code for an access token
+
+.. code:: python
+
+    o = mixcloud.MixcloudOauth(
+        client_id=client_id, client_secret=client_secret,
+        redirect_uri=redirect_uri)
+
+    url = o.authorise_url()
```
Comment (emillon) [inline README.rst]: The method is named `authorize_url`.

**Comment by emillon** [comment]
I'm not very familiar with oauth workflows so I'll have to read a bit on how this is supposed to work :)
But this looks great, thanks a lot for this piece of code.

Do you have some ideas on how it could be tested (ie, the conditions that can make it fail?). Thanks!

File: README.rst (hunk 1 above)
Comment (mrben) [inline README.rst]: Ah, good find, thanks.

**Comment by mrben** [comment]
I'll have a go at writing some tests and also play around rejecting a request to authorize with Mixcloud and see what happens. :)

**Comment by emillon** [comment]
Great, thanks!

**Comment by mrben** [comment]
OK, added tests in now, and test coverage remains at 100% :).

If you want me to squash the commits into a single commit let me know!

**Comment by emillon** [comment]
Thanks for adding tests, I'll merge this as is. I'll be out the next two weeks, but I should be able to upload a pypi release with these changes around the 25th (if you have more features you're welcome!). Feel free to ping me if I forget.

Cheers! :sparkles:

**Comment by mrben** [comment]
Wonderful! :+1:

**Pull request merged**